### PR TITLE
Greatly improve the tooltip component

### PR DESCRIPTION
**Pull request by ivarnakken**
# Description

It is now no longer super buggy!

Firstly it will not overflow out of view when there is no space.
Through the `popper` library, it will find the most suitable postion.
No more need for the `renderDirection` prop and hacky calculations.

Another nice feature is that it will point to the right direction even
when the trigger component is flexed, which it often is. This is most
noticeable on tables.

Some minor styling changes as well, but that's not really the important
part of this change. Worth noting that there is no longer an annoying
buggy line between the arrow and the tooltip whenever there's a bad
render.

# Result

https://github.com/webkom/lego-webapp/assets/69514187/9a7480c8-7a5c-4220-8008-8252171098a0

# Testing

- [x] I have thoroughly tested my changes.

Went through all uses of tooltips. Tried all viewports, hard refreshes and other "stress testing".

---

Resolves https://github.com/webkom/lego/issues/2849

### CODE REVIEW

File: app/components/Tooltip/index.tsx [hunk 1]
```diff
@@ -9,127 +9,72 @@ type Props = {
...
-      renderDirection,
-      pointerPosition,
-    } = this.props;
-    let renderDirectionClass = styles.renderFromCenter;
-    let startPointChildren = 2;
+const Tooltip = ({ children, content, className, style, onClick }: Props) => {
+  const [hovered, setHovered] = useState(false);
+  const tooltipRef = useRef<HTMLDivElement | null>(null);
+  const triggerRef = useRef<HTMLDivElement | null>(null);
+  const arrowRef = useRef<HTMLDivElement | null>(null);
```
Comment (eikhr) [inline app/components/Tooltip/index.tsx:18]:
```suggestion
  const tooltipRef = useRef<HTMLDivElement>(null);
  const triggerRef = useRef<HTMLDivElement>(null);
  const arrowRef = useRef<HTMLDivElement>(null);
```
Don't need the null in type arg here

**Review by eikhr — APPROVED** [review]
I don't love adding more dependencies, but it might be worth it in this case. It works significantly better than before, and the code is way simpler. Looks great!

I think it might be worth keeping the possibility to override the popover position though, f.ex. on the event attendees I think it looks better with the popover over the profile picture.

File: app/components/Tooltip/index.tsx (hunk 1 above)
Comment (ivarnakken) [reply app/components/Tooltip/index.tsx:18]: Good point

**Review by ollfkaih — APPROVED** [review]
Looks good!

**Pull request merged**